src/imitation/data/wrappers.py

```diff
@@ -206,3 +211,124 @@ def step(self, action):
                 "rews": np.stack(self._rews),
             }
         return obs, rew, terminated, truncated, info
+
+
+class HumanReadableWrapper(gym.Wrapper):
+    """Adds human-readable observation to `obs` at every step."""
+
+    def __init__(self, env: Env, original_obs_key: str = "ORI_OBS"):
+        """Builds HumanReadableWrapper.
+
+        Args:
+            env: Environment to wrap.
+            original_obs_key: The key for original observation if the original
+                observation is not in dict format.
+
+        Raises:
+            ValueError: If `env.render_mode` is not "rgb_array".
+
+        """
+        if env.render_mode != "rgb_array":
+            raise ValueError(
+                "HumanReadableWrapper requires render_mode='rgb_array', "
+                f"got {env.render_mode!r}",
+            )
+        self._original_obs_key = original_obs_key
+        super().__init__(env)
+        self._update_obs_space()
+
+    def _update_obs_space(self):
+        # need to reset before render.
+        self.env.reset()
+        example_rgb_obs = self.env.render()
+        new_rgb_space = gym.spaces.Box(
+            low=0,
+            high=255,
+            shape=example_rgb_obs.shape,
+            dtype=np.uint8,
+        )
+        curr_sapce = self.observation_space
+        if isinstance(curr_sapce, gym.spaces.Dict):
+            curr_sapce.spaces[HR_OBS_KEY] = new_rgb_space
+        else:
+            self.observation_space = gym.spaces.Dict(
+                {
+                    HR_OBS_KEY: new_rgb_space,
+                    self._original_obs_key: curr_sapce,
+                },
+            )
+
+    def _add_hr_obs(
+        self,
+        obs: Union[np.ndarray, Dict[str, np.ndarray]],
+    ) -> Dict[str, np.ndarray]:
+        """Adds human-readable observation to obs.
+
+        Transforms obs into dictionary if it is not already, and adds the human-readable
+        observation from `env.render()` under the key HR_OBS_KEY.
+
+        Args:
+            obs: Observation from environment.
+
+        Returns:
+            Observation dictionary with the human-readable data.
+
+        Raises:
+            KeyError: When the key HR_OBS_KEY already exists in the observation
+                dictionary.
+        """
+        if not isinstance(obs, Dict):
+            obs = {self._original_obs_key: obs}
+
+        if HR_OBS_KEY in obs:
+            raise KeyError(f"{HR_OBS_KEY!r} already exists in observation dict")
+        obs[HR_OBS_KEY] = self.env.render()  # type: ignore[assignment]
+        return obs
+
+    def reset(self, **kwargs):
+        obs, info = super().reset(**kwargs)
+        return self._add_hr_obs(obs), info
+
+    def step(self, action):
+        obs, rew, terminated, truncated, info = self.env.step(action)
+        return self._add_hr_obs(obs), rew, terminated, truncated, info
+
+
+def remove_rgb_obs_space(obs_space: gym.Space) -> gym.Space:
```

Comment:
I question if this is the right primitive to expose to users.

An alternative would be to define a policy wrapper or just a policy feature extractor that pulls out the relevant keys from the observation.

Then there's no need to change the observation space or observations at all -- policies are just responsible for filtering what part of the observation stream they attend to. Interactive policy attends to one part, learnable policies attend to the other.

Developer using imitation would have to be responsible for configuring the policies correctly which is a downside of this approach. But developer will need to be thinking about the data collection interface and a bunch of other things anyway so it's not like they'll be able to use an algorithm off-the-shelf anyway, and we can make it somewhat "batteries included" by having some rudimentary support for this in some of our CLI scripts.

WDYT?